crepecake/src/main/java/com/google/cloud/tools/crepecake/blob/BlobHttpContent.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.blob;
+
+import com.google.api.client.http.HttpContent;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.DigestException;
+
+/** {@link Blob}-backed {@link HttpContent}. */
+class BlobHttpContent implements HttpContent {
```

Comment:
Can you explain why this is necessary in the javadoc? I'm assuming it's all about keeping the blob content out of memory?

Reply:
This is an adapter from Blob to HttpContent, which is what is used by the Google HTTP client. This is so that our `Request` class can directly use Blob.